streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamTransformTest.java

```diff
@@ -79,13 +115,13 @@ public void close() {
         KStream<Integer, Integer> stream = builder.stream(topicName, Consumed.with(intSerde, intSerde));
         stream.transform(transformerSupplier).process(processor);
 
-        driver.setUp(builder);
+        driver = new TopologyTestDriver(builder.build(), props, 0L);
         for (int expectedKey : expectedKeys) {
-            driver.process(topicName, expectedKey, expectedKey * 10);
+            driver.pipeInput(recordFactory.create(topicName, expectedKey, expectedKey * 10, 0L));
         }
 
-        driver.punctuate(2);
-        driver.punctuate(3);
+        driver.advanceWallClockTime(2);
+        driver.advanceWallClockTime(1);
```

Comment:
There are no assertion between both calls -- can we do a single call and pass in `3` instead?

Reply:
I think we must because of the punctuation. If we just do a single call and pass 3 a value will be missing from the test results.

Reply:
Ack. Makes sense. Maybe add a comment?

Reply:
The time in this changeset is 1 instead of 3. I imagine you meant to leave it as 3.

Maybe there is some condition you could assert in between these two ticks?